### PR TITLE
chore: Avoid multiple logs messages for the node health controller

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Avoid creating multiple log message on a repair action for an unhealthy node

**How was this change tested?**
- `make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/node/health/controller.go [hunk 1]
```diff
@@ -155,11 +164,25 @@ func (c *Controller) findUnhealthyConditions(node *corev1.Node) (nc *corev1.Node
 }
 
 func (c *Controller) annotateTerminationGracePeriod(ctx context.Context, nodeClaim *v1.NodeClaim) error {
+	if expirationTimeString, exists := nodeClaim.ObjectMeta.Annotations[v1.NodeClaimTerminationTimestampAnnotationKey]; exists {
+		expirationTime, err := time.Parse(time.RFC3339, expirationTimeString)
```
Comment (njtran) [inline pkg/controllers/node/health/controller.go:168]: if we can't parse it, should we override it? Or should we fail close?

File: pkg/controllers/node/health/controller.go [hunk 2]
```diff
@@ -155,11 +164,25 @@ func (c *Controller) findUnhealthyConditions(node *corev1.Node) (nc *corev1.Node
...
+		if err != nil {
+			return err
+		}
+		if c.clock.Now().Before(expirationTime) {
```
Comment (njtran) [inline pkg/controllers/node/health/controller.go:172]: shouldn't this be the opposite? 

if the current time is before the expiration time, we actually do want to annotate it. If it isn't, then we keep it as is. 

Can you add a test for this?

File: pkg/controllers/node/health/controller.go (hunk 1 above)
Comment (engedaam) [reply pkg/controllers/node/health/controller.go:168]: Yeah, it seems reasonable to override it if we can't parse the time.

Comment (njtran) [reply pkg/controllers/node/health/controller.go:168]: Seems like we fail close then if we can't parse it. Can you make that change? The extra call is definitely warranted here

File: pkg/controllers/node/health/controller.go (hunk 2 above)
Comment (engedaam) [reply pkg/controllers/node/health/controller.go]: Yeah, this was the wrong way around, and added a test to validate the behavior

File: pkg/controllers/node/health/controller.go [hunk 3]
```diff
@@ -155,11 +164,24 @@ func (c *Controller) findUnhealthyConditions(node *corev1.Node) (nc *corev1.Node
...
 
 func (c *Controller) annotateTerminationGracePeriod(ctx context.Context, nodeClaim *v1.NodeClaim) error {
+	if expirationTimeString, exists := nodeClaim.ObjectMeta.Annotations[v1.NodeClaimTerminationTimestampAnnotationKey]; exists {
+		expirationTime, err := time.Parse(time.RFC3339, expirationTimeString)
+		if err == nil {
+			if expirationTime.Before(c.clock.Now()) {
+				return nil
+			}
+		}
+	}
```
Comment (njtran) [inline pkg/controllers/node/health/controller.go:174]: nit can combine them
```suggestion
		if err == nil && expirationTime.Before(c.clock.Now()) {
			return nil
		}
	}
```

File: pkg/controllers/node/health/suite_test.go [hunk 4]
```diff
@@ -186,6 +187,23 @@ var _ = Describe("Node Health", func() {
 			nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
 			Expect(nodeClaim.Annotations).To(HaveKeyWithValue(v1.NodeClaimTerminationTimestampAnnotationKey, fakeClock.Now().Format(time.RFC3339)))
 		})
+		It("should not update termination grace period if set past the current time", func() {
```
Comment (njtran) [inline pkg/controllers/node/health/suite_test.go:190]: nit on the wording, seems like you're saying "past" indicating how far in the past it is
```suggestion
		It("should not update termination grace period if set before the current time", func() {
```

**Review by njtran — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**